core/src/test/java/com/linecorp/armeria/client/DnsMetricsTest.java

```diff
@@ -195,24 +196,15 @@ void dns_test_no_answer() throws ExecutionException, InterruptedException {
 
     @Test
     void test_with_real_dns_query() throws ExecutionException, InterruptedException {
-        final MeterRegistry pm1 = PrometheusMeterRegistries.newRegistry();
 
-        try (ClientFactory factory = ClientFactory.builder()
-                .meterRegistry(pm1)
-                .build()) {
+        try (ClientFactory factory = ClientFactory.builder().build()) {
             final WebClient client2 = WebClient.builder()
                     .factory(factory)
                     .build();
 
-            client2.execute(RequestHeaders.of(HttpMethod.GET, "http://wikipedia.com")).aggregate().get();
-            final PrometheusMeterRegistry registry =
-                    (PrometheusMeterRegistry) factory.meterRegistry();
-
-            final double count = registry.getPrometheusRegistry()
-                    .getSampleValue("armeria_client_dns_queries_total",
-                            new String[] {"cause","name","result"},
-                            new String[] {"none","wikipedia.com.", "success"});
-            assertThat(count > 1.0).isTrue();
+            client2.execute(RequestHeaders.of(HttpMethod.GET, "http://google.com")).aggregate().get();
+            final MeterRegistry registry = factory.meterRegistry();
+            System.out.println(MoreMeters.measureAll(registry));
```

Comment:
I do not see any metrics being recorded for this test. Where as if I pass in an external PrometheusMeterRegistries like I did in the other unit tests then I see the counters being incremented. Am I missing something here ?
{armeria.client.connections.lifespan#count{protocol=h2c}=0.0, armeria.client.connections.lifespan#total{protocol=h2c}=0.0, armeria.client.connections.lifespan#max{protocol=h2c}=0.0, armeria.client.connections.lifespan#count{protocol=h1c}=0.0, armeria.client.connections.lifespan#total{protocol=h1c}=0.0, armeria.client.connections.lifespan#max{protocol=h1c}=0.0, armeria.client.dns.queries.written#count{name=google.com.,servers=2001:558:feed:0:0:0:0:1}=0.0, armeria.client.dns.queries#count{cause=none,name=google.com.,result=success}=0.0}

Reply:
No you didn't miss out anything. This is a test case, so we need to specify a concrete meter registry implementation, like you did before. My comment about removing the PrometheusMeterRegistry was for the main sources, not the test sources. Sorry if it was confusing. 🙇

Reply:
Ok. In a real use case how do we pass the meterRegistry down to the ClientFactory? Currently, we have the ClientFactoryBuilder::buildOptions method is where we set the meterRegistry in the DnsResolverGroupBuilder used for the factory. We do something like this 
               builder
              .meterRegistry(ClientFactoryOptions.of().meterRegistry())
              .build(eventLoopGroup);
ClientFactoryOptions.of().meterRegistry() uses the default Metrics.globalRegistry.

Reply:
Ah, of course we should use `ClientFacrotyOptions.of().meterRegistry()` only when a user did not specify one, i.e. should use `options.getOrDefault()`.